packages/cadl-ranch-specs/http/routes/main.tsp

```diff
@@ -0,0 +1,476 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+/**
+ * Define scenario in building the http route/uri
+ */
+@scenarioService("/routes")
+namespace Routes;
+
+@scenario
+@scenarioDoc("""
+  Simple operation at a fixed in an interface
+    Expected path: /routes/fixed
+  """)
+@route("fixed")
+op fixed(): void;
+
+@scenario
+@scenarioDoc("""
+  Simple operation at a fixed in an interface
+    Expected path: /routes/in-interface/fixed
+  """)
+@route("in-interface")
+interface InInterface {
+  @route("fixed")
+  fixed(): void;
+}
+
+@route("path")
+namespace PathParameters {
+  @scenario
+  @scenarioDoc("""
+    Path parameter defined implicitly
+      Value: "a"
+      Expected path: /routes/path/template-only/a
+    """)
+  @route("template-only/{param}")
+  op templateOnly(param: string): void;
+
+  @scenario
+  @scenarioDoc("""
+    Path parameter defined implicitly
+      Value: "a"
+      Expected path: /routes/path/explicit/a
+    """)
+  @route("explicit/{param}")
+  op explicit(@path param: string): void;
+
+  @scenario
+  @scenarioDoc("""
+    Path parameter annotated with @path but not defined explicitly in the route
+      Value: "a"
+      Expected path: /routes/path/annotation-only/a
+    """)
+  @route("annotation-only")
+  op annotationOnly(@path param: string): void;
+
+  @route("reserved-expansion")
+  namespace ReservedExpansion {
+    @scenario
+    @scenarioDoc("""
+      Defines a path parameter that shouldn't encode reserved characters. It should however still encode the other url characters.
+        Param value: "foo/bar baz"
+        Expected path: "/routes/path/reserved-expansion/template/foo/bar%20baz"
+      """)
+    @route("template/{+param}")
+    op template(param: string): void;
+
+    @scenario
+    @scenarioDoc("""
+      Defines a path parameter that shouldn't encode reserved characters. It should however still encode the other url characters.
+        Param value: "foo/bar baz"
+        Expected path: "/routes/path/reserved-expansion/annotation/foo/bar%20baz"
+      """)
+    @route("annotation")
+    op annotation(@path(#{ allowReserved: true }) param: string): void;
+  }
+
+  @route("simple")
+  namespace Simple {
```

Comment:
should maybe update all those namespace to be `XExpansion`

Reply:
more like should it be `SimpleExpansion` `ReservedExpansion` `PathExpansion` (the terms used in the Rfc)